internal/monitors/jaegergrpc/monitor.go

```diff
@@ -0,0 +1,168 @@
+package jaegergrpc
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"github.com/jaegertracing/jaeger/proto-gen/api_v2"
+	"github.com/pkg/errors"
+	"github.com/signalfx/signalfx-agent/internal/core/common/constants"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/jaegergrpc/jaegerprotobuf"
+	"github.com/signalfx/signalfx-agent/internal/monitors/types"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+)
+
+const gracefulShutdownTimeout = time.Second * 5
+
+var logger = utils.NewThrottledLogger(log.WithFields(log.Fields{"monitorType": monitorType}), 30*time.Second)
+
+func init() {
+	monitors.Register(&monitorMetadata, func() interface{} { return &Monitor{} }, &Config{})
+}
+
+// TLSCreds is a configuration struct for specifying a cert file and key file for tls
+type TLSCreds struct {
+	// The cert file to use for tls
+	CertFile string `yaml:"certFile"`
+	// The key file to use for tls
+	KeyFile string `yaml:"keyFile"`
+}
+
+func (tls *TLSCreds) Credentials() (credentials.TransportCredentials, error) {
+	var creds credentials.TransportCredentials
+	var err error
+	if tls != nil {
+		creds, err = credentials.NewServerTLSFromFile(tls.CertFile, tls.KeyFile)
+	}
+
+	return creds, err
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `yaml:",inline" acceptsEndpoints:"false" singleInstance:"true"`
+	// The host:port on which to listen for traces.
+	ListenAddress string `yaml:"listenAddress" default:"0.0.0.0:14250"`
+	// TLS are optional tls credential settings to configure the GRPC server with
+	TLS *TLSCreds `yaml:"tls,omitempty"`
+}
+
+// Monitor that accepts and forwards SignalFx data
+type Monitor struct {
+	Output types.Output
+	grpc   *grpc.Server
+	ln     net.Listener
+}
+
+var _ api_v2.CollectorServiceServer = (*Monitor)(nil)
+
+func extractRemoteAddressToContext(ctx context.Context) (net.IP, bool) {
+	var sourceIP net.IP
+
+	// get peer connection info from grpc context
+	p, hasSource := peer.FromContext(ctx)
+	if !hasSource || p.Addr.String() == "" {
+		return nil, false
+	}
+
+	// separate host from port
+	host, _, err := net.SplitHostPort(p.Addr.String())
+	if err != nil {
+		return nil, false
+	}
+
+	// parse ip from host
+	sourceIP = net.ParseIP(host)
+	if sourceIP == nil {
+		return nil, false
+	}
+
+	return sourceIP, true
+}
+
+// PostSpans implements the jeager api_v2.CollectorServiceServer interface.  The grpc server will pass the jaeger
+// batches it receives to this method.  This method will convert the jaeger batches to SignalFx spans and pass them
+// on to the output writer.
+func (m *Monitor) PostSpans(ctx context.Context, r *api_v2.PostSpansRequest) (*api_v2.PostSpansResponse, error) {
+	if r != nil {
```

Comment:
It is a little clearer to just return the empty response (or maybe some error since getting a nil request seems abnormal) if `r == nil` instead of nesting all of the core logic of the function in this if statement.